addon/components/horizontal-bar-chart.js

```diff
@@ -431,37 +440,112 @@ const HorizontalBarChartComponent = ChartComponent.extend(FloatingTooltipMixin,
+    const rightMostValueLabelWidth = this._getElementWidthForValue(valueLabelElements, maxValue);
+
+    const padding = (2 * this.get('labelPadding')) + this.get('yAxisTitleHeightOffset') + 5;
+    const outerWidth = this.get('outerWidth');
+
+    const width = outerWidth - leftMostValueLabelWidth - rightMostValueLabelWidth - padding;
+    const xScale = this._xScaleForWidth(width);
+
+    const maxNegativeBarWidth = Math.abs(xScale(minValue) - xScale(0));
+    const maxPositiveBarWidth = Math.abs(xScale(maxValue) - xScale(0));
+
+    // If the sum of the widths of the longest bar in a direction and its value label is larger
+    // than the longest grouping label on the same side of the chart, then the relevant width on
+    // that side is the width of the value label
+    var leftWidth = leftMostValueLabelWidth;
+    // In the case where the left grouping label is wider than the sum of the largest left bar
+    // and its value label, the goal is to find the distance between the left edge of the chart
+    // and the end of the left bar.
+    if (maxLeftGroupingLabelWidth > maxNegativeBarWidth + leftMostValueLabelWidth) {
+      // The amount of space to the left of the axis is fixed at the width of the grouping label.
+      // The amount of space to the right of most positive bar is fixed at the width of the
+      // value label for that bar. Knowing this, we can compute the width of the positive bar.
+      let realPositiveBarWidth = outerWidth - maxLeftGroupingLabelWidth - rightMostValueLabelWidth - padding;
+      // From the positive bar width, we can compute the negative bar width
+      let realNegativeBarWidth = this._mostNegativeBarWidth(realPositiveBarWidth);
+      leftWidth = maxLeftGroupingLabelWidth - realNegativeBarWidth;
+    }
 
-      const [leftWidth, rightWidth] = [leftLabels, rightLabels].map((elements) => {
-        return this._maxWidthOfElements(elements);
-      });
-      return {
-        left: leftWidth,
-        right: rightWidth
-      };
+    // This is the inverse of the logic above used for leftWidth
+    var rightWidth = rightMostValueLabelWidth;
+    if (maxRightGroupingLabelWidth > maxPositiveBarWidth + rightMostValueLabelWidth) {
+      let realNegativeBarWidth = outerWidth - maxRightGroupingLabelWidth - leftMostValueLabelWidth - padding;
+      let realPositiveBarWidth = this._mostPositiveBarWidth(realNegativeBarWidth);
+      rightWidth = maxRightGroupingLabelWidth - realPositiveBarWidth;
     }
+
+    return {
+      left: leftWidth,
+      right: rightWidth
+    };
+  },
+
+  /*
+   * For charts whith a mix of positive and negative values, given the width of
+   * the most positive bar, get the width of the most negative bar. The ratio
+   * of the widths of the two bars is the same as the ratio between the min and
+   * max values
+   * @private
+   * @param {Number} mostPositiveBarWidth
+   * @return {Number}
+   */
+  _mostNegativeBarWidth: function(mostPositiveBarWidth) {
```

Comment:
having a function's name without using verb is weird when it's not a computed property.